database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/UserViewModel.swift

```diff
@@ -127,43 +115,49 @@ class UserViewModel: ObservableObject {
       self.isLoading.toggle()
     }
 
-    let userID = getuid()
-    guard let key = postRef.childByAutoId().key else { return }
-    let post = ["uid": userID,
-                "author": Auth.auth().currentUser?.email,
-                "title": title,
-                "body": body]
-    let childUpdates = ["/posts/\(key)": post,
-                        "/user-posts/\(String(describing: userID))/\(key)/": post]
-    ref.updateChildValues(childUpdates)
+    if let userID = Auth.auth().currentUser?.uid {
+      let postListRef = ref.child("posts")
+      guard let key = postListRef.childByAutoId().key else { return }
+      let post = ["uid": userID,
+                  "author": Auth.auth().currentUser?.email,
+                  "title": title,
+                  "body": body]
+      let childUpdates = ["/posts/\(key)": post,
+                          "/user-posts/\(userID)/\(key)/": post]
+      ref.updateChildValues(childUpdates)
+    }
 
     // end loading animation
     withAnimation {
       self.isLoading.toggle()
     }
   }
 
-  func fetchPosts() {
-    // read data by listening for value events
-    refHandle = postRef.observe(DataEventType.value, with: { snapshot in
-      // retrieved data is of type dictionary of dictionary
-      guard let value = snapshot.value as? [String: [String: Any]] else { return }
-      // sort dictionary by keys (most to least recent)
-      let sortedValues = value.sorted(by: { $0.key > $1.key })
-      // store content of sorted dictionary into "posts" variable
-      self.posts = sortedValues.compactMap { Post(id: $0, dict: $1) }
-    })
+  func getPosts(tabOpened: String) {
```

Comment:
Make `tabOpened` an enum with a raw value of String for better case-handling below.